### PR TITLE
chore(ci): bumping which github action version we use for node.

### PR DESCRIPTION
**Pull request by JessicaSachs**
chore(ci): bumping which github action version we use for node.

This is so that we can use the correct Volta version specified inside of the
package.json when we're running tests and building our project.

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally for affected output targets
- [ ] Tests (`npm test`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

We're using a different version of node when building and shipping our packages than we use when we develop.

Issue URL:

## What is the new behavior?

We keep the development version (specified by Volta) synchronized with the version of node we're executing CI in.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by JessicaSachs** [comment]
Wait on #327 before reviewing.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,17 +14,18 @@ jobs:
...
+      - name: Checkout Code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # 3.5.2
+      - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # 3.6.0
         with:
-          node-version: 16.x
-      - uses: actions/checkout@v1
+          node-version-file: './package.json'
       - name: Restore Dependency Cache
-        uses: actions/cache@v3
+        uses: actions/cache@2b250bc32ad02700b996b496c14ac8c2840a2991
```
Comment (rwaskiewicz) [inline .github/workflows/ci.yml:28]: Based on https://github.com/actions/cache/releases, this appears to be for a patch release of 2.18.x. Could we `88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8` use instead?
```suggestion
        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # 3.3.1
```
Screenshot 2023-04-26 at 4 45 15 PM

File: .github/workflows/actions/publish-npm.yml [hunk 2]
```diff
@@ -17,9 +17,9 @@ inputs:
 runs:
   using: 'composite'
   steps:
-    - uses: actions/setup-node@v3
+    - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # 3.6.0
```
Comment (rwaskiewicz) [inline .github/workflows/actions/publish-npm.yml:20]: Normally, we prefix these little comments with a 'v' (e.g. `# v3.6.0`). 

It's probably not a big deal, I know that Dependabot/Renovate (neither of which are in this repo ATM) are smart enough to bump the version comment as well as the SHA ([like so](https://github.com/ionic-team/stencil/pull/4266/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721L14)). No action required here, just wanted to get that thought out for future considerations

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
All in all, output looks good!

Screenshot 2023-04-26 at 4 50 16 PM

I had one blocking comment to fix a versioning downgrade that snuck in. I'd like to fix that now, as it uses Node 12 based dependencies (which I think are being sunset in GHA in the next month or so).

**Review by sean-perkins — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (JessicaSachs) [reply .github/workflows/ci.yml]: Nice. Thanks for the catch.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -14,17 +14,18 @@ jobs:
...
+      - name: Checkout Code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
         with:
-          node-version: 16.x
-      - uses: actions/checkout@v1
+          node-version-file: './package.json'
       - name: Restore Dependency Cache
-        uses: actions/cache@v3
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.5.1
```
Comment (rwaskiewicz) [inline .github/workflows/ci.yml:28]: Nit
```suggestion
        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
```

**Review by rwaskiewicz — APPROVED** [review]

**Pull request merged**